### PR TITLE
Added --fail-after-error option to exit gulp process after linter error.

### PR DESCRIPTION
**Pull request by stefanpl**
Right now, grunt will always exit cleanly with an exit code of 0. This is due to the usage of grunt-plumber. 

While this is handy during development, it prevents CI(-like) setups from checking if there are any errors in the gulp tasks.

### CODE REVIEW

File: tasks/scripts.js [hunk 1]
```diff
@@ -8,7 +10,7 @@ module.exports = (gulp, $, pkg) => {
       default: { concat: true },
     }), args);
     return gulp.src(pkg.gulpPaths.scripts.src)
-      .pipe($.plumber())
+      .pipe(process.env.GULP_ABORT_ON_ERROR ? noop() : $.plumber())
```
Comment (sun) [inline tasks/scripts.js]: This appears to be the same as the built-in errorHandler option – do we need the additional dependency?
https://www.npmjs.com/package/gulp-plumber#user-content-optionserrorhandler

```suggestion
      .pipe($.plumber({ errorHandler: !process.env.GULP_ABORT_ON_ERROR }))
```

Comment (stefanpl) [reply tasks/scripts.js]: You are right @sun, the additional dependency is not needed. I was event able to remove plumber altogether, the same behavior can be achieved by using the options for gulp-stylelint and gulp-eslint. Please see my new commit.

File: README.md [hunk 2]
```diff
@@ -31,6 +31,15 @@ Commonly used Gulp tasks shared across projects.
 - `gulp fractal:start`: Start Fractal server (if config path exists in `package.json`)
 - `gulp fractal:build`: Build static version of Fractal project
 
+## Abort on errors
+
+In the default setup, all tasks are continued even if any of the task produces an error. Furthermode, calling `grunt […]` will always exit with code 0.
```
Comment (LucaPipolo) [inline README.md]: In the default setup, all tasks are continued even if any of the task produces an error. Furthermode, calling `gulp […]`
```suggestion
In the default setup, all tasks are continued even if any of the task produces an error. Furthermode, calling `gulp […]` will always exit with code 0.
```

Comment (LucaPipolo) [reply README.md]: @colourgarden — can you double check English here? 🇬🇧

File: tasks/scripts.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ module.exports = (gulp, $, pkg) => {
       default: { concat: true },
     }), args);
     return gulp.src(pkg.gulpPaths.scripts.src)
-      .pipe($.plumber())
+      .pipe($.if(process.env.GULP_ABORT_ON_ERROR, $.eslint.failAfterError()))
```
Comment (fabianmarz) [inline tasks/scripts.js]: Currently we're using `minimist` to retrieve CLI options while hardly rely on environment variables. Does it makes sense to adjust the other options to use env variables as well by introducing [`dotenv`](https://www.npmjs.com/package/dotenv), at least for the `options.production` condition? 🤔

File: README.md (hunk 2 above)
Comment (colourgarden) [inline README.md]:
```suggestion
In the default setup, all tasks are continued even if one of the tasks produces an error. Furthermore, calling `gulp […]` will always exit with code 0.
```

File: tasks/scripts.js (hunk 3 above)
Comment (fabianmarz) [reply tasks/scripts.js]: Based on the comment by @sun we could add an option like this and check if it's set. If not, the pipe will break otherwise.

```
$ gulp build --continue

const options = Object.assign($.minimist(process.argv.slice(2), {
  string: ['outputStyle'],
  boolean: ['sourcemaps', 'production', 'continue'],
  default: { },
}), args);

.pipe($.if(options.continue, $.plumber()))
```

A Gulp task will fail if an error occurs. [Here's](https://gist.github.com/floatdrop/8269868) a small article about the behaviour.

We also need to introduce it for the scripts task as this is not handled by this PR yet.

Comment (sun) [reply tasks/scripts.js]: This sounds like the proper solution to me. Can we make `--continue` default to `true` via `package.json`?

Or change the suggested change here to explicitly check for `options.continue === false`?

Comment (stefanpl) [reply tasks/scripts.js]: Just pushed two new commits, trying to integrate the proposed changes. I totally agree with @fabianmarz  that having flags AND an env value is odd, so I introduced an error handling flag `fail-after-error`. 
Also, having understood now what plumber does thanks to the article, I put it back where it belongs.

**What about `--continue`?**
As @sun pointed out, that's a flag to control gulp behavior. If there is a sequence of tasks (like in our `build` task, by default one of them failing will prevent the others from being run, unless `--continue` is provided. 
The way I see it, using the same flag name to control the error behavior of the linters (like `gulp scripts --continue=false`) would be highly confusing, as you could end up with something like `gulp --continue build --continue=false`. That's why I proposed to name our flag `--fail-after-error`. 

**Summing up:**
- The proposed change is backwards compatible. No flag – behavior does not change.
- `--fail-after-error` flag will make scripts and styles task exit with error code > 0 if they detect an error.
- documented in readme.md

Looking forward to getting your feedback!

File: README.md (hunk 2 above)
Comment (stefanpl) [reply README.md]: Thanks! Fixed it, please see the latest version of readme.md

File: tasks/scripts.js (hunk 3 above)
Comment (sun) [reply tasks/scripts.js]: Happy to move forward with this for now. We can still revise it in a future version if necessary.  @fabianmarz What do you think?

Only the docs could use some tweaks, but I will try to add a fixup for those directly now.

File: tasks/scripts.js [hunk 4]
```diff
@@ -4,13 +4,14 @@ module.exports = (gulp, $, pkg) => {
   // @task: Build JS from components.
   const task = (args = {}) => {
     const options = Object.assign($.minimist(process.argv.slice(2), {
-      boolean: ['concat', 'sourcemaps', 'production'],
+      boolean: ['concat', 'sourcemaps', 'production', 'fail-after-error'],
       default: { concat: true },
     }), args);
     return gulp.src(pkg.gulpPaths.scripts.src)
-      .pipe($.plumber())
+      .pipe($.if( ! options['fail-after-error'], $.plumber()))
```
Comment (sun) [inline tasks/scripts.js]:
```suggestion
      .pipe($.if(!options['fail-after-error'], $.plumber()))
```

File: tasks/styles.js [hunk 5]
```diff
@@ -26,12 +26,13 @@ module.exports = (gulp, $, pkg) => {
   const task = (args) => {
     const options = Object.assign($.minimist(process.argv.slice(2), {
       string: ['outputStyle'],
-      boolean: ['sourcemaps', 'production'],
+      boolean: ['sourcemaps', 'production', 'fail-after-error'],
       default: { },
     }), args);
     return gulp.src(pkg.gulpPaths.styles.src, { base: pkg.gulpPaths.styles.srcDir })
-      .pipe($.plumber())
+      .pipe($.if( ! options['fail-after-error'], $.plumber()))
```
Comment (sun) [inline tasks/styles.js]:
```suggestion
      .pipe($.if(!options['fail-after-error'], $.plumber()))
```

File: tasks/scripts.js (hunk 3 above)
Comment (fabianmarz) [reply tasks/scripts.js]: Agree, can be merged :)
